engine/src/XmlLoader.java

```diff
@@ -55,6 +56,91 @@ public void checkValidXml() throws XmlException {
         checkHeadPointer();
     }
 
+    public void loadRepo(){
+        RepositoryManager repositoryManager = new RepositoryManager();
+        repositoryManager.createNewRepository(magitRepository.getLocation());
+        // TODO verify
+        MagitSingleCommit commit = null;
+        for(MagitSingleBranch magitSingleBranch: magitBranches.getMagitSingleBranch()){
```

Comment:
what should that for loop do ? 
move it to a function that explain its action , so it will be easier to know the purpose of the code with just a quick look